Source/VMware.vSphereDSC/DSCResources/VMHostService.ps1

```diff
@@ -0,0 +1,155 @@
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        $this.UpdateVMHostService($vmHost)
+    }
+
+    [bool]Test() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+
+        return !$this.ShouldUpdateVMHostService($vmHost)
+    }
+
+    [VMHostService]Get() {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vmHostService = [VMHostService]::new()
+        $vmHostService.Server = $this.Server
+
+        $this.ConnectVIServer()
+        $vmHost = $this.GetVMHost()
+        $this.PopulateResult($vmHost, $vmHostService)
+
+        return $vmHostService
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns a boolean value if the VMHostService needs to be updated.
+    #>
+    [bool]ShouldUpdateVMHostService($vmHost) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vmHostCurrentService = Get-VMHostService -Server $this.Connection -VMHost $vmHost | Where-Object {$_.Key -eq $this.Key}
+
+        $shouldUpdateVMHostService = @()
+        $shouldUpdateVMHostService += $this.Policy -ne $vmHostCurrentService.Policy
+        $shouldUpdateVMHostService += $this.Running -ne $vmHostCurrentService.Running
+
+        return ($shouldUpdateVMHostService -Contains $true)
+    }
+
+    <#
+    .DESCRIPTION
+
+    Updates the configuration of the VMHostService
+    #>
+    [void] UpdateVMHostService($vmHost) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vmHostCurrentService = Get-VMHostService -Server $this.Connection -VMHost $vmHost | Where-Object {$_.Key -eq $this.Key}
+
+        if ($vmHostCurrentService.Policy -ne $this.Policy) {
+            Set-VMHostService -HostService $vmHostCurrentService -Policy $this.Policy.ToString() -Confirm:$false
+        }
+
+        if ($vmHostCurrentService.Running -ne $this.Running) {
+            if ($vmHostCurrentService.Running) {
+                Stop-VMHostService -HostService $vmHostCurrentService -Confirm:$false
+            }
+            else {
+                Start-VMHostService -HostService $vmHostCurrentService -Confirm:$false
+            }
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Populates the result returned from the Get() method with the values of the VMHostService.
+    #>
+    [void] PopulateResult($vmHost, $vmHostService) {
+        Write-Verbose -Message "$(Get-Date) $($s = Get-PSCallStack; "Entering {0}" -f $s[0].FunctionName)"
+
+        $vmHostCurrentService = Get-VMHostService -Server $this.Connection -VMHost $vmHost | Where-Object {$_.Key -eq $this.Key}
+        $vmHostService.Key = $vmHostCurrentService.Key
```

Comment:
Add this line here: $vmHostService.Name = $vmHostCurrentService.Name.

Reply:
You need to retrieve the Name from the server to verify that a successful connection was made.